src/NUnitEngine/nunit.engine/Services/RuntimeFrameworkService.cs

```diff
@@ -211,6 +211,20 @@ private static void ApplyImageData(TestPackage package)
                         requiresAssemblyResolver = true;
                     }
                 }
+
+                if (frameworkName == null)
+                {
+                    foreach (var reference in module.AssemblyReferences)
+                        if (reference.Name == "mscorlib" && BitConverter.ToUInt64(reference.PublicKeyToken, 0) == 0xac22333d05b89d96)
+                        {
+                            // We assume 3.5, since that's all we are supporting
+                            // Could be extended to other versions if necessary
+                            // Format for FrameworkName is invented - it is not
+                            // known if any compilers supporting CF use the attribute
+                            frameworkName = ".NETCompactFramework,Version=3.5";
+                            break;
+                        }
+                }
```

Comment:
If any future compilers use TargetFrameworkAttribute for CF, then this code won't run as frameworkName will be non null. Use of the PublicKeyToken is documented for all known versions of CF. Ideally, we would distinguish various versions of CF here, but I didn't do so. The Version part of the string is not used for anything.